Improve event page documentation and examples (#28411)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/runtime/onmessage/index.md

browser.runtime.onMessage.addListener(handleMessage);
```

> > **Warning:** Do not prepend `async` before the function. That changes the meaning to [sending an asynchronous response using a promise](#sending_an_asynchronous_response_using_a_promise), which would effectively be the same as `sendResponse(true)`.

Reviewer: ```suggestion
> **Warning:** Do not prepend `async` to the function. That changes the meaning to [sending an asynchronous response using a promise](#sending_an_asynchronous_response_using_a_promise), which is effectively the same as `sendResponse(true)`.
```

### Sending an asynchronous response using a Promise

> > **Note:** Promise as a return value is not supported in Chrome until [Chrome bug 1185241](https://crbug.com/1185241) is resolved. As an alternative, [return true and use sendResponse](#sending_an_asynchronous_response_using_sendresponse).

Reviewer: ```suggestion
> **Note:** Promise as a return value is not supported in Chrome until [Chrome bug 1185241](https://crbug.com/1185241) is resolved. As an alternative, [return true and use `sendResponse`](#sending_an_asynchronous_response_using_sendresponse).
```
